File: LeetCode/000/73/LC73SetZero.js
{
    /**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
*/

    var setZeroes = function (matrix) {

        let m = matrix.length;
        let n = matrix[0].length;

        let zeroRow = new Set();
        let zeroCol = new Set();


        for (let i = 0; i < m; i++) {

            for (let j = 0; j < n; j++) {

                if (matrix[i][j] === 0) {
                    zeroRow.add(i); // storing index 
                    zeroCol.add(j); // storing index 
                }
            }
        }

        zeroRow.forEach(row => {
            for (let j = 0; j < n; j++) {
                matrix[row][j] = 0;
            }
        })

        zeroCol.forEach(col => {
            for (let i = 0; i < m; i++) {
                matrix[i][col] = 0;
            }
        })

    };
}

File: DSA/Leetcode/LeetcodeProblem/0000/73/LC73SetZero.java

class Solution {
    public void setZeroes(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;

        HashSet<Integer> hs_col = new HashSet<>();
        HashSet<Integer> hs_row = new HashSet<>();

        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[i].length; j++) {
                if (matrix[i][j] == 0) {
                    hs_row.add(i);
                    hs_col.add(j);
                }
            }
        }

        for (int col : hs_col) {
            for (int k = 0; k < m; k++) {
                matrix[k][col] = 0;
            }
        }

        for (int row : hs_row) {
            for (int k = 0; k < n; k++) {
                matrix[row][k] = 0;
            }
        }
    }
}
